napari/layers/labels/labels.py

```diff
@@ -374,24 +387,21 @@ def color(self, color):
         self.color_mode = color_mode
 
     def _ensure_int_labels(self, data):
-        """Ensure data is integer by converting from float if required"""
+        """Ensure data is integer by converting from bool if required, raising an error otherwise"""
         looks_multiscale, data = guess_multiscale(data)
         if not looks_multiscale:
             data = [data]
-        if any(np.issubdtype(d.dtype, np.floating) for d in data):
-            warnings.warn(
-                "Float dtypes are not supported for Labels layers. Converting data to integers..."
-            )
-            int_data = []
-            for d in data:
-                if d.dtype == np.float64:
-                    int_data.append(d.astype(np.int64))
-                elif d.dtype == np.float32:
-                    int_data.append(d.astype(np.int32))
-                else:
-                    int_data.append(d)
-            data = int_data
-
+        int_data = []
+        for d in data:
+            if np.issubdtype(d.dtype, np.floating):
+                raise TypeError(
```

Comment:
Great work @DragaDoncila 

Now that we implemented localization, we need to make sure include `trans._` on the new strings we introduce, since these errors will appear on the notifications manager.

you will need to import `trans` if not already imported and use it like:

```python

from napari.utils.translation import trans

raise TypeError(trans._("Only integer types are supported for Labels layers, but data contains {dtype}.").format(dtype=d.dtype)
```

---

I opened this as a reminder :) 
https://github.com/napari/napari/pull/2497

Reply:
Thanks for catching that @goanpeca !